### DISCUSSION TITLE
KeyError: 'role_82b410c922aa1308001b2cf4617646c7'

### DISCUSSION THREAD

**Discussion by judev-jbg** [saleweaver/python-amazon-sp-api#566] [General]
Hello:
I am trying to retrieve the offer elements of an asin with get_item_offers but I get the following error:
`Traceback (most recent call last):
  File "C:\Junior.marketing\AppData\Local\Programs\Python\Python38\librium-packages\cachetools\__init__.py", line 72, in __getitem__. 
    return self.__data[key]
KeyError: 'role_82b410c922aa1308001b2cf4617646c7'`
*I am sending the credentials by code through a client (dict).
`myCredentials = dict(
    refresh_token=st.conf_api_amz["refresh_token"],
    lwa_app_id=st.conf_api_amz["lwa_app_id"],  
    lwa_client_secret=st.conf_api_amz["lwa_client_secret"],  
    aws_access_key=st.conf_api_amz["aws_access_key"],  
    aws_secret_key=st.conf_api_amz["aws_secret_key"], 
    role_arn=st.conf_api_amz["role_arn"])
)`
Script:
`products = Products(credentials=myCredentials, marketplace=Marketplaces.ES)
print(products.get_item_offers(asin='B00G706ZFQ', item_condition="New"))`

Note: I have verified that when using https://sp-api-playground.saleweaver.com/api/product-pricing-api-model/productPricingV0.json it does work since I do receive a response from the offers of that ASIN, however, through python I can't. 

Can you help me?

**Comment by saleweaver** [comment]
If it works in the playground, the only thing I can think of is that you have some mixup in the credentials in code.
